pkg/logs/wait_for_image.go

```diff
@@ -123,20 +124,28 @@ func (w *imageWaiter) waitBuild(ctx context.Context, writer io.Writer, namespace
 		}
 	}()
 
-	event, err := watchTools.ListWatchUntil(ctx,
-		&listAndWatchBuild{kpackClient: w.KpackClient, namespace: namespace, buildName: buildName},
-		filterErrors(func(event watch.Event) (bool, error) {
-			build, ok := event.Object.(*v1alpha1.Build)
-			if !ok {
-				return false, errors.New("unexpected object received")
-			}
-
-			return !build.Status.GetCondition(corev1alpha1.ConditionSucceeded).IsUnknown(), nil
-		}))
+	targetBuild, err := w.KpackClient.KpackV1alpha1().Builds(namespace).Get(ctx, buildName, v1.GetOptions{})
 	if err != nil {
 		return "", err
 	}
 
+	event := &watch.Event{Object: targetBuild}
+	if targetBuild.Status.GetCondition(corev1alpha1.ConditionSucceeded).IsUnknown() {
```

Comment:
Would it work to just send an initialResourceVersion of "0" to avoid fetching the initial build?

Reply:
this is what we were considering: https://github.com/kubernetes/client-go/blob/master/tools/watch/until.go#L97

Reply:
It seems like that entire doc comment is out-of-date and it was unclear if it now supported "O". Did we test it?

Reply:
@matthewmcnew  To answer your initial question no, it would not work. But we refactored to build out our own version of listWatchUntil to address this.